### PR TITLE
Matrix comparisons

### PR DESCRIPTION
**Pull request by Kieran-Fishwick-TfN**
# Changes

Added matrix comparisons and allowed multi TLDs to be calculated within MatrixReport

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases)
- [x] Has documentation (including user guide) been updated
- [] Have new dependencies been added (or changed) in relevant `requirements.txt` NA
- [x] Code linting, tests and other workflows pass

----
📚 Documentation preview 📚: https://caftoolkit--170.org.readthedocs.build/en/170/

### CODE REVIEW

**Review by MattBuckley-TfN — CHANGES_REQUESTED** [review]

**Comment by MattBuckley-TfN** [comment]
Accidentally missed typing the overall comment for the above review.

I think all the logic looks good so my comments are just around tidying up some of the code, shouldn't take too long to address them. Let me know if you have any questions about them

File: src/caf/toolkit/pandas_utils/matrices.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ def __init__(
         self._matrix = matrix
         self._translated_matrix: pd.DataFrame | None = None
         self._describe: pd.DataFrame | None = None
-        self._distribution: cost_utils.CostDistribution | None = None
+        self._distribution: pd.DataFrame | None = None
```
[the remark this answers is not in the document]
Comment (Kieran-Fishwick-TfN) [reply src/caf/toolkit/pandas_utils/matrices.py:52]: Because CostDistribution doesn't support MultiTLDs
MultiTLDs are currently only implemented in caf.distribute. Maybe one for the back log to migrate them here?

File: src/caf/toolkit/pandas_utils/matrices.py [hunk 2]
```diff
@@ -176,10 +263,15 @@ def sector_matrix(self) -> pd.DataFrame | None:
...
     @property
-    def distribution(self) -> cost_utils.CostDistribution | None:
+    def distribution(self) -> pd.DataFrame | None:
         """Distribution if `trip_length_distribution` has been called, otherwise none."""
         return self._distribution
 
+    @property
+    def vkms(self) -> pd.Series | None:
+        """Vehicle kms if `calc_vehicle_kms` has been called, otherwise none."""
+        return self._vkms
```
[the remark this answers is not in the document]
Comment (Kieran-Fishwick-TfN) [reply src/caf/toolkit/pandas_utils/matrices.py:273]: I would agree but
 ```if matrix_report.vkms is not None```
is used below to determine matrix reports have vkms when comparing which would give a warning during expected behaviour.
